src/main/java/thecodex6824/thaumcraftfix/mixin/event/PlayerEventsMixin.java

```diff
@@ -52,4 +57,41 @@ private static Object[] filterCategories(Object[] originalCats) {
 	return retained.toArray(new String[retained.size()]);
     }
 
+	@Inject(
+			method = "getRunicCharge(Lnet/minecraft/item/ItemStack;)I",
+			at = @At("HEAD"),
+			remap = false,
+			cancellable = true
+	)
+	private static void modifyRunicCharge(ItemStack stack, CallbackInfoReturnable<Integer> cir) {
+		int base = 0;
+		if (stack.hasTagCompound() && stack.getTagCompound().hasKey("TC.RUNIC")) {
+			base += stack.getTagCompound().getInteger("TC.RUNIC");
```

Comment:
I think it might be possible to use a [MixinExtras expression](https://github.com/LlamaLad7/MixinExtras/wiki/Expression-Language#binary-expressions) to target the addition itself and then add the difference (if any) between the byte and integer values, which would then mean the entire method doesn't have to be overridden. That might be complex to do though, so if you can't do it I can do it later.